packages/clerk-js/src/ui/elements/Modal.tsx

```diff
@@ -17,17 +17,18 @@ type ModalProps = React.PropsWithChildren<{
   contentSx?: ThemableCssProp;
   containerSx?: ThemableCssProp;
   canCloseModal?: boolean;
+  outsidePress?: boolean;
   style?: React.CSSProperties;
 }>;
 
 export const Modal = withFloatingTree((props: ModalProps) => {
-  const { handleClose, handleOpen, contentSx, containerSx, canCloseModal, id, style } = props;
+  const { handleClose, handleOpen, contentSx, containerSx, canCloseModal, outsidePress = true, id, style } = props;
   const { disableScroll, enableScroll } = useScrollLock(document.body);
   const overlayRef = useRef<HTMLDivElement>(null);
   const { floating, isOpen, context, nodeId, toggle } = usePopover({
     defaultOpen: true,
     autoUpdate: false,
-    outsidePress: e => e.target === overlayRef.current,
+    outsidePress: outsidePress === true ? e => e.target === overlayRef.current : false,
```

Comment:
is this different from the `canCloseModal` property?

Reply:
`canCloseModal` prevents closing the modal entirely, close icon is removed to. `canCloseModal` is used within the `useDismiss` hook. This is also an internal only prop.

Reply:
Thank you for explaining. Initially, I thought it would be a little confusing having both of these properties at the same time.

Reply:
Yeah, no problem, I can see how the naming can be confusing. Still noodling on names, let me know if you have opinions https://github.com/clerk/javascript/pull/5139#issuecomment-2659831105

Reply:
I've also went ahead and added comments to clarify the different functionality https://github.com/clerk/javascript/pull/5139/commits/7155cb4cea76e1fead60e13b589f014839c76fae